File: javascript/sintaxe-basica.js
/*
==============================================================
                    Revisão Basica JavaScript
==============================================================
*/


// ====== Declaração de variavel ======
let name = "Hackathon";   // pode ser reatribuida;
const ano = 2026;         //não pode ser reatribuida;
// ano = 2027 -> daria erro!

// ====== Strings e Concatenações ======
let tech = "IA";         
console.log(`Vamos usar ${tech} no projeto`);

// ====== Arrays ======
let linguagens = ["JS", "Python", "Java"];

// Adicionar ao final
linguagens.push("C++");

//Tamanho do Array
console.log(linguagens.length);

//Pegar o primeiro item
console.log(linguagens[0]);

// Retirar do final
linguagem_removida = linguagens.pop(); // retirou C++ e guardou na variavel.

// Retirar o primeiro
linguagens.shift();

// ====== Condicionais ======
let nota = 8;

if (nota >= 9){
    console.log("Excelente!");
}else if(nota >= 6){
    console.log("Bom");
}else{
    console.log("Reprovado!");
}

// ====== Loops ======
let nomes = ["Geovanna", "Amora", "Nicolas", "Messi"];

// Loop for classico
for( let i = 0; i < nomes.length; i++){
    console.log(nomes[i]);
}

//Loop for moderno
for( let nome of nomes){
    console.log(nome);
}

// While
let contador = 0;
while(contador < 3){
    console.log(contador);
    contador ++;
}

// ====== Funções ======

// função tradicional;
function somar (a,b){
    return a + b;
}

// Arrow Function
const multiplicar = (a,b) => a * b;

console.log(somar(2,3));

File: python/sintaxe-basica.py

#==============================================================
#                    Revisão Basica Python
#==============================================================

# ====== Declaração de Variavel ======
nome = "Hackathon" # Cria e já atribui
ano = 2026         # Python não tem "constantes" reais, mas por convenção usamos MAIÚSCULAS para indicar (ex: ANO = 2026)

# ====== Strings e Concatenação ======
tech = "IA"
print(f"Vamos usar {tech} no projeto!")

# ====== Listas (Python) ======
linguagens = ["JS", "Python", "Java"]

# Adicionar ao final
linguagens.append("C++")

# Tamanho da lista
print(len(linguagens))

# Pegar o primeiro item
print(linguagens[0])

# Remover Ultimo
linguagens = ["JS", "Python", "Java", "C++"]

# Remove o último item ("C++") e guarda na variável
linguagem_removida = linguagens.pop()

print(linguagens)         # Resultado: ['JS', 'Python', 'Java']
print(linguagem_removida) # Resultado: C++

# ====== Condicionais ======
nota = 8

if nota >= 9:
    print("Excelente")
elif nota >= 7:
    print("Bom")
else:
    print("Precisa melhorar")

# ====== Loops ======
nomes = ["Grazielle", "Isabela", "Luana"]

# Loop For (muito mais direto)
for nome in nomes:
    print(nome)

# Se precisar do índice, use range() ou enumerate()
for i in range(len(nomes)):
    print(nomes[i])

# While
contador = 0
while contador < 3:
    print(contador)
    contador += 1 # Python não tem "++"

# ====== Funções ======
# Função tradicional
def somar(a, b):
    return a + b

print(somar(2, 3))